File: android/src/main/java/headset/events/headsetStateChange/HeadsetState.java
package headset.events.headsetStateChange;

public enum HeadsetState {
  //TODO: REMOVE TEST
  TEST(-1),
  INIT(0),
  CONNECTING(1),
  CONNECTED(2),
  WORKING(3),
  STOPPED(4),
  DISCONNECTED(5),
  COMPLETE(6),
  CONNECTION_FAILED(100),
  RECORDING_START(7),
  RECORDING_END(8),
  ERROR(101),
  GET_DATA_TIME_OUT(9),
  CHECK_SUM_FAIL(1000),
  RECORD_FAIL(1001);

  private int value;

  HeadsetState(int value) {
    this.setValue(value);
  }
  
  public int getValue() {
    return value;
  }

  public void setValue(int value) {
    this.value = value;
  }

  public static HeadsetState fromValue(int value) {
    for (HeadsetState e : values()) {
      if (e.value == value) {
        return e;
      }
    }
    return null;
  }
}

File: lib/enums/headset_state.dart
/// Represents the different states of the MindWave Mobile 2 headset.
///
/// This enum defines various states that the MindWave Mobile 2 headset can be in
/// during initialization, connection, operation, and error conditions.
enum HeadsetState {
  /// Initial state.
  INIT(0),

  /// The headset is in the process of connecting.
  CONNECTING(1),

  /// The headset is successfully connected.
  CONNECTED(2),

  /// The headset is working, receiving data.
  WORKING(3),

  /// The headset is stopped.
  STOPPED(4),

  /// The headset is disconnected.
  DISCONNECTED(5),

  /// The connection process is complete.
  COMPLETE(6),

  /// Connection failed.
  CONNECTION_FAILED(100),

  /// Recording started.
  RECORDING_START(7),

  /// Recording ended.
  RECORDING_END(8),

  /// Error occurred.
  ERROR(101),

  /// Timeout occurred while waiting for data.
  GET_DATA_TIME_OUT(9),

  /// Checksum failed.
  CHECK_SUM_FAIL(1000),

  /// Recording failed.
  RECORD_FAIL(1001);

  /// The integer value associated with the state.
  final int value;

  /// Constructs a [HeadsetState] instance with the given integer value.
  const HeadsetState(this.value);

  /// Retrieves a [HeadsetState] instance from its integer value.
  ///
  /// - [value]: The integer value of the desired headset state.
  ///
  /// Returns the corresponding [HeadsetState] instance.
  factory HeadsetState.fromValue(int value) =>
      values.firstWhere((element) => element.value == value);
}
